jina/peapods/zmq/__init__.py

```diff
@@ -305,10 +358,23 @@ async def send_message(self, msg: 'Message', sleep: float = 0, **kwargs):
         :param kwargs: keyword arguments
         """
         # await asyncio.sleep(sleep)  # preventing over-speed sending
-        try:
-            num_bytes = await send_message_async(
-                self.out_sock, msg, **self.send_recv_kwargs
+        if self.args.dynamic_out_routing:
+            await self.send_message_dynamic(msg)
+            return
+        else:
+            self._send_message_via(self.out_sock, msg)
+
+    async def send_message_dynamic(self, msg: 'Message'):
+        for routing_graph, out_sock in self.get_dynamic_next_routes(msg):
+            new_message_proto = msg.proto
+            new_message_proto.envelope.targets.CopyFrom(routing_graph.proto)
+            await self._send_message_via(
```

Comment:
Just to clarify my understanding. This is `asynchronous` but until the `_send_message_via` returns the next one cannot be sent right? 

Does it need to be awaited? Can't all of these results put in different futures and await them after the `for-loop`?

Reply:
Yes, that might be an optimization in the future.

Reply:
`asyncio.create_task(blah())` should work for fire & forget.

Reply:
just want to make sure if we are aware that this `for loop` will actually be executed sequentially

Reply:
Changed it. Please double check, if it as you imagine.